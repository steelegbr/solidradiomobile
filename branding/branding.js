/**
 * Provides branding functions.
 */

import { DefaultTheme, DarkTheme } from 'react-native-paper';

/**
 * Generates a station theme.
 * @param {bool} darkMode An indication if we are in dark mode.
 * @param {station} station The station this theme is for.
 */

export function generateStationTheme(darkMode, station) {

    let theme = generateTheme(darkMode);
    theme.colors.primary = station.primary_colour;
    theme.colors.accent = station.primary_colour;
    theme.mode = 'adaptive';
    return theme;

}

/**
 * Generates a theme with no station.
 * @param {bool} darkMode An indication if we are in dark mode.
 */

export function generateTheme(darkMode) {

    if (darkMode) {
        return {
            ...DarkTheme,
            roundness: 10,
            colors: {
                ...DarkTheme.colors
            }
        };
    } else {
        return {
            ...DefaultTheme,
            roundness: 10,
            colors: {
                ...DefaultTheme.colors
            }
        };
    }

}
