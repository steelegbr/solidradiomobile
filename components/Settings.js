import React from 'react';
import { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { Title, Switch, Paragraph } from 'react-native-paper';
import { connect } from 'react-redux';
import { setDarkMode, setHighBitrate } from '../reducers/actions';

class Settings extends Component {

    render() {

        const { darkMode, highBitrate, styles, setDarkMode, setHighBitrate } = this.props;

        return(
            <View >
                <Title style={styles.header}>Settings</Title>
                <View style={styles.row}>
                    <Paragraph>Dark Mode</Paragraph>
                    <Switch value={darkMode} onValueChange={() => setDarkMode(!darkMode)} testID="darkModeToggle" />
                </View>
                <View style={styles.row}>
                    <Paragraph>Stream High Bitrate</Paragraph>
                    <Switch value={highBitrate} onValueChange={() => setHighBitrate(!highBitrate)} testID="highBitrateToggle" />
                </View>
            </View>
        );
    }

}

const stylesTemplate = StyleSheet.create({
    row: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingVertical: 8,
        paddingHorizontal: 16
    },
    header: {
        padding: 16
    }
});

const mapStateToProps = state => {
    return {
        darkMode: state.settings.darkMode,
        highBitrate: state.settings.highBitrate,
        styles: {
            ...stylesTemplate,
            view: {
                backgroundColor: state.theme.colors.background
            }
        }
    };
};

const mapDispatchToProps = dispatch => {
    return {
        setDarkMode: mode => dispatch(setDarkMode(mode)),
        setHighBitrate: mode => dispatch(setHighBitrate(mode))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Settings);