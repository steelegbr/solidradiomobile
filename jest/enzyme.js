/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import 'react-native';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

/**
 * Set up Enzyme to mount to DOM, simulate events,
 * and inspect the DOM in tests.
 */
Enzyme.configure({ adapter: new Adapter() });

// JSDOM - required for Enzyme's mount

const { JSDOM } = require('jsdom');
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;
global.window = window;
global.document = window.document;