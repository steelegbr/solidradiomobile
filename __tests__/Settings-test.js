/**
 * Tests the settings screen.
 */

import React from 'react';
import Settings from '../components/Settings';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { generateTheme } from '../branding/branding';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

// Redux

const mockStore = configureStore([]);

// Tests

it('renders-off-off', () => {

    const store = mockStore({
        theme: generateTheme(false),
        settings: {
            darkMode: false,
            highBitrate: false
        }
    });

    const settings = renderer.create(
        <Provider store={store}>
            <Settings />
        </Provider>
    ).toJSON();

    expect(settings).toMatchSnapshot();

});

it('renders-off-on', () => {

    const store = mockStore({
        theme: generateTheme(false),
        settings: {
            darkMode: false,
            highBitrate: true
        }
    });

    const settings = renderer.create(
        <Provider store={store}>
            <Settings />
        </Provider>
    ).toJSON();

    expect(settings).toMatchSnapshot();

});

it('renders-on-off', () => {

    const store = mockStore({
        theme: generateTheme(true),
        settings: {
            darkMode: true,
            highBitrate: false
        }
    });

    const settings = renderer.create(
        <Provider store={store}>
            <Settings />
        </Provider>
    ).toJSON();

    expect(settings).toMatchSnapshot();

});

it('renders-on-on', () => {

    const store = mockStore({
        theme: generateTheme(true),
        settings: {
            darkMode: true,
            highBitrate: true
        }
    });

    const settings = renderer.create(
        <Provider store={store}>
            <Settings />
        </Provider>
    ).toJSON();

    expect(settings).toMatchSnapshot();

});

it('dark-mode-change', () => {

    // Arrange

    const originalDarkMode = true;

    let store = mockStore({
        theme: generateTheme(originalDarkMode),
        settings: {
            darkMode: originalDarkMode,
            highBitrate: false
        }
    });

    const settings = mount(
        <Provider store={store}>
            <Settings />
        </Provider>
    );

    // Act

    settings.find('[testID="darkModeToggle"]').first().props().onValueChange();

    // Assert

    const expectedActions = [ { type: 'SET_DARK_MODE', mode: !originalDarkMode } ];
    expect(store.getActions()).toStrictEqual(expectedActions);

});

it('high-bitrate-change', () => {

    // Arrange

    const originalHighBitrate = true;

    let store = mockStore({
        theme: generateTheme(false),
        settings: {
            darkMode: false,
            highBitrate: originalHighBitrate
        }
    });

    const settings = mount(
        <Provider store={store}>
            <Settings />
        </Provider>
    );

    // Act

    settings.find('[testID="highBitrateToggle"]').first().props().onValueChange();

    // Assert

    const expectedActions = [ { type: 'SET_HIGH_BITRATE', mode: !originalHighBitrate } ];
    expect(store.getActions()).toStrictEqual(expectedActions);

});