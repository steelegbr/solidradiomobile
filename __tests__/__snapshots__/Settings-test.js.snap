// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders-off-off 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
        ],
      ]
    }
  >
    Settings
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Dark Mode
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="darkModeToggle"
      tintColor=""
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Stream High Bitrate
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="highBitrateToggle"
      tintColor=""
      value={false}
    />
  </View>
</View>
`;

exports[`renders-off-on 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
        ],
      ]
    }
  >
    Settings
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Dark Mode
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="darkModeToggle"
      tintColor=""
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Stream High Bitrate
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="highBitrateToggle"
      tintColor=""
      value={true}
    />
  </View>
</View>
`;

exports[`renders-on-off 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
        ],
      ]
    }
  >
    Settings
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Dark Mode
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="darkModeToggle"
      tintColor=""
      value={true}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Stream High Bitrate
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="highBitrateToggle"
      tintColor=""
      value={false}
    />
  </View>
</View>
`;

exports[`renders-on-on 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "padding": 16,
            },
          ],
        ],
      ]
    }
  >
    Settings
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Dark Mode
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="darkModeToggle"
      tintColor=""
      value={true}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Stream High Bitrate
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#03dac4"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="highBitrateToggle"
      tintColor=""
      value={true}
    />
  </View>
</View>
`;
